client/starwhale/core/runtime/model.py

```diff
@@ -1475,7 +1545,33 @@ def restore(
         run_with_progress_bar("runtime restore...", operations)
 
     @staticmethod
-    def _install_dependencies(
+    def _install_dependencies_with_runtime_yaml(
+        workdir: Path,
+        runtime_yaml: t.Any,
+        env_dir: t.Union[Path, str],
+        skip_deps: t.Optional[t.List[DependencyType]] = None,
+    ) -> None:
+        env_dir = Path(env_dir)
+
+        mode = runtime_yaml.get("mode", PythonRunEnv.VENV)
+        configs = runtime_yaml.get("configs", {})
+        skip_deps = skip_deps or []
+        console.print(
+            f":baby_bottle: install runtime.yaml dependencies @ {env_dir} for lock..."
+        )
+        deps_config = Dependencies(runtime_yaml.get("dependencies", []))
+        for dep in deps_config.deps:
+            if dep.kind in skip_deps:
+                logger.debug(f"skip {dep} to install")
+                continue
+
+            _func = (
+                dep.conda_install if PythonRunEnv.CONDA == mode else dep.venv_install
```

Comment:
What will happen when "use current system shell"?
Will `venv_install` make unexpected behavior?

Reply:
when users config `env_use_shell=True`:
1. get prefix path by [env](https://github.com/star-whale/starwhale/pull/1450/files#diff-368402d5a9ece27e6cf34e94a5028e75d935c7fc0712dd10f4868800c749c780R1252) 
2. use prefix path bin to install pkgs.